packages/inter-protocol/src/proposals/addAssetToVault.js

```diff
@@ -204,7 +214,65 @@ export const registerScaledPriceAuthority = async (
   );
 };
 
-/** @typedef {import('./econ-behaviors.js').EconomyBootstrapPowers} EconomyBootstrapPowers */
+// wait a short while after end to allow things to settle
+const BUFFER = 5n * 60n;
+// let's insist on 20 minutes leeway for running the scripts
+const COMPLETION = 20n * 60n;
+
+// If there is a liveSchedule, 1) run now if start is far enough away,
+// otherwise, 2) run after endTime.
+// If neither liveSchedule nor nextSchedule is defined, 3) run now.
+// If there is only a nextSchedule, 4) run now if startTime is far enough away,
+// else 5) run after endTime
+const whenQuiescent = async (schedules, timer, thunk) => {
+  const { nextAuctionSchedule, liveAuctionSchedule } = schedules;
+  const now = await E(timer).getCurrentTimestamp();
+
+  const waker = Far('addAssetWaker', { wake: () => thunk() });
+
+  if (liveAuctionSchedule) {
+    const safeStart = TimeMath.subtractAbsRel(
+      liveAuctionSchedule.startTime,
+      COMPLETION,
+    );
+
+    if (TimeMath.compareAbs(safeStart, now) < 0) {
+      // case 2
```

Comment:
this case comment is helpful. please include for all the cases.

alternately, can you factor this so the case is determined in one phase and then a switch executes the appropriate action?